packages/Toggle/src/Toggle.js

```diff
@@ -1,7 +1,352 @@
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
-import { ccClassName } from '@leafygreen-ui/lib';
-import * as style from './style';
+import { createDataProp, emotion } from '@leafygreen-ui/lib';
+import { colors } from '@leafygreen-ui/theme';
+
+const { css, cx } = emotion;
+
+const toggleInput = createDataProp('toggle-input');
+const toggleGroove = createDataProp('toggle-groove');
+
+const transitionInMS = 150;
+
+const inputStyle = css`
+  margin: 0;
+  position: absolute;
+  left: 100%;
+  top: 100%;
+  pointer-events: none;
+  opacity: 0;
+`;
+
+// We use a div for the focus state rather than a pseudo-element
+// because said pseudo-element would need to be on the label element
+// which can't use the contained input's focus pseudo-class.
+const focusStateStyle = css`
+  transition: all ${transitionInMS}ms ease-in-out;
+  position: absolute;
+  top: -2px;
+  bottom: -2px;
+  left: -2px;
+  right: -2px;
+  border: 2px solid rgba(67, 177, 229, 0.25);
+  border-radius: 50px;
+  opacity: 0;
+  transform: scale(0.8);
+
+  ${toggleInput.selector}:focus ~ & {
```

Comment:
I think we have a different way of thinking about styles.

I would have instead inverted this. Made a `createDataProp('focus-state')` and inside of the toggleInput styles done `&:focus ~ ${focusState.selector}`. i.e. keep the ampersands on the left hand side. Trying to explain my thinking more, but wanted to get this written out for now.

Reply:
The way I think about styles is to apply styles for the same element as close as possible. The problem with `&:focus ~ ${focusState.selector}` is that it's not styling the element whose selector you're nested in. Rather than keeping the ampersand to the left, `${toggleInput.selector}:focus ~ &` keeps related styles closer together. It's a very conscious method I've used for a while now that helps readability IMO.